libyara/proc/linux.c

```diff
@@ -46,6 +46,8 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <yara/proc.h>
 #include <yara/strutils.h>
 
+#define _FILE_OFFSET_BITS 64
```

Comment:
This does not work, it must be defined before any include for it to have an effect.

This can be confirmed by:

- removing the AC_SYS_LARGEFILE
- adding a `_Static_assert(sizeof(off_t) == 8);`
- compiling with `-m32`

the static assert fails, indicating the scanning would be invalid.

That's why i was suggesting adding a static assert, it is very easy to break the fix without realizing it.